runtime/src/chpl-mem.c

```diff
@@ -93,15 +93,15 @@ int chpl_posix_memalign(void** ptr, size_t alignment, size_t size) {
 
 void* chpl_valloc(size_t size)
 {
-  return chpl_memalign(chpl_getHeapPageSize(), size);
+  return chpl_memalign(chpl_getSysPageSize(), size);
```

Comment:
What's the reason for this change? If we're aligning an allocation, it seems like we want to align to the heap page size, not the system page size. Note that we distinguish between the two because sometimes the heap has a different page size than the system page size. An example is when using the Cray native comm layer with hugepages.